### PR TITLE
begin to integrate it analysis data prep into pipeline

### PR DESCRIPTION
**Pull request by galengorski**
This is my first attempt at organizing the IT analysis functions to get them integrated in a pipeline framework, so we might have to iterate to find a good solution here. The way I was thinking of setting it up is with `it_analysis_data_prep.py` script with input from the `it_analysis_data_prep_config.yaml` file where: 

1. sources and sinks are selected based on column names
2. the number of lag days is selected
3. those sink and lagged sources are written to 03_it_analysis/src/out file as a pickle file

Those lagged sources and sinks are in lists (one item for each site/model run) and they are to be read in by the next set of functions that will include calculating the correlations and plotting the heat maps. I think the way I'll initially set it up will be to use the sinks and sources for correlation, mutual information, and transfer entropy each with their own config.yaml file, and have each function output a similar matrix to be used for heatmap plotting.

### CODE REVIEW

File: 03_it_analysis/src/it_analysis_data_prep.py [hunk 1]
```diff
@@ -112,39 +111,40 @@ def lag_sources(n_lags, srcs_list):
...
+    #create out directory if it doesn't already exist
+    os.makedirs(out_dir, exist_ok = True)
+    #write snks_list to file 
+    snks_file = open(out_dir+'snks', "wb")
+    pickle.dump(snks_list, snks_file)
+    snks_file.close()
+    #write srcs_list_lagged to file 
+    srcs_file = open(out_dir+'srcs_lagged', "wb")
+    pickle.dump(srcs_list_lagged, srcs_file)
+    srcs_file.close()
```
Comment (amsnyder) [inline 03_it_analysis/src/it_analysis_data_prep.py:146]: Could these be included in the select_sinks() and lag_sources() functions? Or perhaps create a new function that takes in a list of lists and file names and saves the output? Just trying to think about how to keep the main() function as concise as possible

File: 03_it_analysis/src/make_correlation_matrix.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    srcs_lagged_pickle = config['srcs_lagged_pickle']
+    #select the sinks
+    snks_list_pickle = config['snks_list_pickle']
+    #select the out_directory
+    out_dir = config['out_dir']
+    corr_matrix = create_correlation_matrix(srcs_lagged_pickle, snks_list_pickle)
+    
+    corr_matrix_save = open(out_dir+'corr_matrix', "wb")
+    pickle.dump(corr_matrix, corr_matrix_save)
+    corr_matrix_save.close()
```
Comment (amsnyder) [inline 03_it_analysis/src/make_correlation_matrix.py:44]: Same question about saving in another function applies here

**Review by amsnyder — CHANGES_REQUESTED** [review]
Sorry I missed this pull request! I think this looks great overall. Just a question about trying to move the list saving into a function.

I am also wondering what you think of merging the 3 configs into one config for the step? We can still reference only the variables needed for a script because they are grouped with the script names as headers. There isn't really a problem with the way it is now, but that's how we set up the fetch step.

I think one additional step is to incorporate these into the Snakefile, but we could do that after merging this pull request!

File: 03_it_analysis/src/it_analysis_data_prep.py (hunk 1 above)
Comment (galengorski) [reply 03_it_analysis/src/it_analysis_data_prep.py:146]: Yeah, this is a good idea, I'll move the save calls from `main()` into their specific functions. Good idea.

**Pull request merged**